### ISSUE TITLE
AnimationPlayer track keys can not be loaded again after first loading with ResourceLoader

### ISSUE THREAD

**Issue by Skyfish1** [godotengine/godot#10101]
**Operating system or device - Godot version:**
Windows 7, Godot Godot_v2.1.3-stable_win64

**Issue description:**
AnimationPlayer track keys can not be loaded again after first loading with ResourceLoader.
 
Situation: We are in an **in-game** Level Editor, not in Godot Editor!

After modifying a previously loaded scene and loading it again, modified AnimationPlayer track keys are not loaded instead they keep their modified value even if the whole scene (including the AnimationPlayer) is queued_free and completely reloaded from disk.

Expected:
Animation track keys (which are part of the loaded Scene) are are the ones in the loaded Scene not the ones in memory of the queued_free scene.

Details:
I chose GDScript modified scenes (.tscn) as a save format, so I can also use the Godot Editor to tweak the Levels. 

Loading:
`playable_map = load(path).instance()`

I also tried to disabling caching:
`playable_map = ResourceLoader.load(path,"",true).instance()`

**Steps to reproduce:**
Following the steps in the attached project to reproduce the bug
- Load saved map
- change track keys
- Load map again

-> changes to track keys are not loaded, but other changes to other nodes are.

**Link to minimal example project:**
[RessourceLoaderAnimPlayerBug.zip](https://github.com/godotengine/godot/files/1201690/RessourceLoaderAnimPlayerBug.zip)

**Comment by Skyfish1** [comment]
Same Issue with Godot 2.1.4 and Curve2D. 

Steps:
1st I save a Level to tscn file on Disk:
`var packed_scene = PackedScene.new()
packed_scene.pack(tscn)
ResourceSaver.save(path, packed_scene)`

Then I run it with get_tree().change_scene().

Then I modify the Level (clear_points() on Curve2D and add Points) and save it again overwrite the file. 

Then I run it again with get_tree().change_scene().

Result:
The OLD! Curve2D is shown. 

Expected: 
Changes are loaded


I checked the saved File in an text editor: Its correct!
I exited Godot, restarted and loaded the Level: It's correct!

There is some kind of caching going on that refuses to update some Objects (Curve2D) if the a scene of same name was loaded before. If I save with different file every time (not overwrite) it's also correct.
